### ISSUE TITLE
Anub'rekhan not linked to his Crypt Guards.

### ISSUE THREAD

**Issue by Pugnadeus** [TrinityCore/TrinityCore#20714]
**Description:**

If I am inside Naxxramas, I can pull one of Anub'rekhan's Crypt Guards without aggroing the other Crypt Guard or Anub'rekhan himself. 

**Current behaviour:**

It is possible to aggro only Anub'rekhan or one of the Crypt Guards if we aggro them from the side.

**Expected behaviour:**

Pulling any of the Crypt Guards or Anub'rekhan himself will result in all of them getting aggroed.

**Steps to reproduce the problem:**

1. CHANGEME Step 1 include entries of affected creatures / items / quests with a link to the relevant wowhead page.  
2. Step 2
3. Step 3

**Branch(es):**

3.3.5 master

**TC rev. hash/commit:** 

TrinityCore rev. 607034064f04 2017-10-07 17:13:34 +0200 (3.3.5 branch) (Win64, RelWithDebInfo, Static) (authserver)

**TDB version:**  TDB_full_world_335.63_2017_04_18

**Operating system:** Windows 10

**Comment by Killyana** [comment]
Is it normal, Crypt Guards appears only on 25man?

**Comment by Pugnadeus** [comment]
So I should post the top one which you said will work in the querry, or will it be updated in the official TC?

**Comment by Pugnadeus** [comment]
One more thing @Killyana, the query you added does work if Anub'rekhan is pulled first, however it does not work if his Crypt Guards are pulled first, I can still pull each one of them from the side, kill them and then engage the boss without adds at the start.

**Comment by tkrokli** [comment]
Maybe @Wyreth would like to pick up the challenge to see if he can find out what can be done about that action in the SAI core source?

**Comment by Pugnadeus** [comment]
It would be incredible if someone could start working on this issue, since it creates a lot of problem in MC and sub-raids on my server which is  hosting vanilla as end-game content, is there a forum where I can ask some more complicated questions about TC?

**Comment by lachtanek** [comment]
Was digging into this issue earlier, call for help only works when creature has a victim so its kinda buggy for some reason when used with aggro event (didnt have time to check why, through)
You can use SET_IN_COMBAT_WITH_ZONE (38) with target CREATURE_GUID (10), but it may be a bit of a hack.

**Comment by Killyana** [comment]
The easiest way is to delay the call of help by 1 secs.

**Comment by lachtanek** [comment]
@Killyana I think thats even more hacky solution than mine tho :D

**Comment by Wyreth** [comment]
@lachtanek maybe not - it might sound strange, but in some of my almost expired memories from WotLK retail I recall creatures waiting 1-2 seconds before answering a call for help :P

**Comment by lachtanek** [comment]
@Wyreth alright, fair enough, still weird tho

**Comment by Killyana** [comment]
Yes it makes sense the text is displayed then after 1-2 secs npcs will enter combat.

**Comment by Keader** [comment]
Just put call of combat with boss, when Crypt Guards enter in combat
(you can do it using EnterCombat(Unit* who) in a ScriptedAI)

**Comment by Killyana** [comment]
Yes but in this case the SAI action mut be fixed.

**Comment by Pugnadeus** [comment]
Will an issue like this be fixed in the 3.3.5 branch or will I have to do this manually?

**Comment by tkrokli** [comment]
Ideally it will be fixed in the TC source (3.3.5 / master), but it depends on someone coming up with a working solution.

**Comment by lachtanek** [comment]
One solution would be to add second optional parameter victim to Creature::CallForHelp. If noone comes up with better solution, I will open a PR with it.

**Comment by offl** [comment]
Why nobody talks about formations? :D

**Comment by lachtanek** [comment]
Sometimes you can't put creatures into formations like when they already are in on or, in this case, they're temporary summons

**Comment by offl** [comment]
That's why I had a feeling that I am stupid and missed something, because four days and nobody's talking about formations.. there must be a reason.

**Comment by Pugnadeus** [comment]
I'm new to Trinity, but how many people are working on it?

**Comment by ccrs** [comment]
Since working implies, by definition, some sort of revenue, 0

**Comment by tkrokli** [comment]
Yes, it all depends on how you define "working" or what you actually want to know.
There is a certain number of members in the TrinityCore organization, but some are inactive
and some are more active than others. And like ccrs said, none of them receive a salary.

**Comment by Treeston** [comment]
If you wanna pay me a full-time job I'll gladly work on this silly project.

Until then, you get fixes when they happen.

**Comment by Pugnadeus** [comment]
Apologies for the attitude in the past @Treeston , I didn't want to come off that way, I'm very grateful for the fix!
